### ISSUE TITLE
Protobuf doesn't build under `--all_incompatible_changes` yet Bazel does

### ISSUE THREAD

**Issue by brandjon** [bazelbuild/bazel#6689]
The latest tagged version of protobuf is v3.6.1, but this version does not build under `--all_incompatible_changes` due to `--incompatible_package_name_is_a_function` failures in it's `//protobuf.bzl`. Despite this, I see that we currently depend on v3.6.1 via a `new_local_repository` rule.

I'm not familiar enough with our use of protobuf to understand why we're still able to build Bazel itself with `--incompatible_package_name_is_a_function`. Does this indicate a bug in `new_local_repository`, that the imported repo isn't being subjected to the same incompatible flags?

**Comment by brandjon** [comment]
I don't know that I would call for a release of protobuf just for this issue alone. There will always be new incompatible changes in Bazel, at least until we have a slower release cycle.

**Comment by benjaminp** [comment]
Bazel doesn't use `protobuf.bzl` because it just contains rules and macros for generating protobufs that are superseded by the builtin aspects.
